metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -387,6 +398,12 @@ public void replay(PartitionChangeRecord record) {
         String topicPart = topicInfo.name + "-" + record.partitionId() + " with topic ID " +
             record.topicId();
         newPartitionInfo.maybeLogPartitionChange(log, topicPart, prevPartitionInfo);
+        if ((newPartitionInfo.leader != newPartitionInfo.preferredReplica()) && 
```

Comment:
Can we have a function like "hasPreferredLeader" on `PartitionControlInfo`, to make this simpler?

Reply:
Can do this in a follow-on